include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,686 @@
+{
+  return fe_face_values_neighbor == nullptr;
+}
+
+
+
+template <int dim, int spacedim>
+std::vector<types::global_dof_index>
+FEInterfaceValues<dim, spacedim>::get_interface_dof_indices() const
+{
+  return interface_dof_indices;
+}
+
+
+
+template <int dim, int spacedim>
+std::array<unsigned int, 2>
+FEInterfaceValues<dim, spacedim>::interface_dof_to_dof_indices(
+  const unsigned int interface_dof_index) const
+{
+  AssertIndexRange(interface_dof_index, n_current_interface_dofs());
+  return dofmap[interface_dof_index];
+}
+
+
+
+template <int dim, int spacedim>
+const FEFaceValuesBase<dim, spacedim> &
+FEInterfaceValues<dim, spacedim>::get_fe_face_values(
+  const unsigned int cell_index) const
+{
+  AssertIndexRange(cell_index, 2);
+  Assert(
+    cell_index == 0 || !at_boundary(),
+    ExcMessage(
+      "You are on a boundary, so you can only ask for the first FEFaceValues object."));
+
+  return (cell_index == 0) ? *fe_face_values : *fe_face_values_neighbor;
+}
+
+
+
+template <int dim, int spacedim>
+Tensor<1, spacedim>
+FEInterfaceValues<dim, spacedim>::normal(const unsigned int q_point_index) const
+{
+  return fe_face_values->normal_vector(q_point_index);
+}
+
+
+
+template <int dim, int spacedim>
+double
+FEInterfaceValues<dim, spacedim>::shape_value(
+  const bool         use_upstream_value,
+  const unsigned int interface_dof_index,
+  const unsigned int q_point,
+  const unsigned int component) const
+{
+  const auto dofpair = dofmap[interface_dof_index];
+
+  if (use_upstream_value && dofpair[0] != numbers::invalid_unsigned_int)
+    return get_fe_face_values(0).shape_value_component(dofpair[0],
+                                                       q_point,
+                                                       component);
+  if (!use_upstream_value && dofpair[1] != numbers::invalid_unsigned_int)
+    return get_fe_face_values(1).shape_value_component(dofpair[1],
+                                                       q_point,
+                                                       component);
+
+  return 0.0;
+}
+
+
+
+template <int dim, int spacedim>
+double
+FEInterfaceValues<dim, spacedim>::jump(const unsigned int interface_dof_index,
+                                       const unsigned int q_point,
+                                       const unsigned int component) const
+{
+  const auto dofpair = dofmap[interface_dof_index];
+
+  if (dofpair[1] == numbers::invalid_unsigned_int)
+    return get_fe_face_values(0).shape_value_component(dofpair[0],
+                                                       q_point,
+                                                       component);
+  if (dofpair[0] == numbers::invalid_unsigned_int)
+    return -get_fe_face_values(1).shape_value_component(dofpair[1],
+                                                        q_point,
+                                                        component);
+  return 0.0; // jump is zero for continuous shape functions
```

Comment:
I don't think this is right. You are thinking that because the shape function appears to live on both cells, that it is continuous and so the . But that's not true. Think about the situation of an RT element: Its *normal* component is continuous, but its tangential component is not, even though it lives on both sides of the cell. I think you really have to compute the difference here.